Automattic/jetpack
# Add the Package_Version_Tracker class

Add the new `Package_Version_Tracker` class, which will obtain the package versions using the package versions using the `jetpack_package_versions` filter.

#### Changes proposed in this Pull Request:
* Add the new `Package_Version_Tracker` class. This class obtains the package versions and determines whether the package versions have changed. If the versions have changed, it updates an option and sends the updated versions to WPCOM.
* Add the `jetpack_package_versions` filter, which the tracker class will use to get the package versions.
* Use the `jetpack_package_versions` filter in the Connection, Sync, and Backup packages to send the package versions to the tracker class.

#### Jetpack product discussion
p9dueE-3b1-p2

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:
1. Checkout this branch. Activate and connect Jetpack.
2. Check that the `jetpack_package_versions` option is populated with the Backup, Connection, and Sync versions: `wp option get jetpack_package_versions`
3. Using your wpcom sandbox, check that your site's `jetpack_package_versions` option is correct.
4. You can test that package versions are updated by changing the package version values in the `Package_Version` class in the [Connection](https://github.com/Automattic/jetpack/pull/20365/files#diff-ec835716664525f9a699acafb4900028163167f44c4ad831467b728c1534a1d9R15), [Sync](https://github.com/Automattic/jetpack/pull/20365/files#diff-8cbe6de8fca27651a9743f3cfb6ad21f466db46d01fb42f0637e13216e0e1373R15), and [Backup](https://github.com/Automattic/jetpack/pull/20365/files#diff-fc6ceea571e3f430150e93340e84470cddb4c491ad4e2e7817e40bbfafd767a7R15) packages.
